docs/api.md

```diff
@@ -456,6 +468,97 @@ router.route('/post/:id')
   });
 ```
 
+### router.use()
+
+```js
+router.use(fn?): this
+```
+
+| Param | Type | Description |
+| --- | :-: | --- |
+| `fn` | (req, res) => void | A middleware function. |
+
+Attaches middleware functions to current router.
+
+_Middleware_ are functions that pre-process <a href="#request" jump-to-id="request">`Request`</a> objects and <a href="#response" jump-to-id="response">`Reponse`</a> objects before they are sending to routing handlers.
+
+You can add **top-level middleware** to current <a href="#server" jump-to-id="server">`Server`</a> object:
+
+```js
+// server
+const server = createServer();
+
+server.use((req, res) => {
+  res.headers.set('X-Server', 'Mocker');
+});
+
+server.router.get('/greet', 'Hello world');
+
+// client
+fetch('/greet').then((res) => {
+  console.log(res.headers.get('X-Server')); // > 'Mocker'
+});
+```
+
+Or mount **scoped middleware** to specific <a href="#router" jump-to-id="router">`Routers`</a>:
+
+```js
+// server(routing handlers are omitted)
+router.scope('/sub')
+  .use((req, res) => {
+    res.headers.set('X-Server', 'Mocker');
+  });
+
+// client
+fetch('/yooo').then((res) => {
+  console.log(res.headers.get('X-Server')); // > null
+});
+
+fetch('/sub/yeah').then((res) => {
+  console.log(res.headers.get('X-Server')); // > 'Mocker'
+});
+```
+
+You can even add **asynchronous middleware** by returning a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise):
+
+```js
+// server(routing handlers are omitted)
+router.use(async (req, res) => {
+  await asyncStuff();
+  res.headers.set('X-Middleware', 'Async');
+});
+
+// client
+fetch('/path').then((res) => {
+  console.log(res.headers.get('X-Middleware')); // > 'Async'
+});
+```
+
+Middleware can be inherited from upstream:
```

Comment:
I mean change the sentence `Middleware can be inherited from upstream:` to `Upstream middleware are inherited:`. 😅